src/http/ngx_http_wasm_api.c

```diff
@@ -918,8 +918,14 @@ ngx_http_wasm_req_get_header(ngx_http_request_t *r, char *key,  int32_t key_size
                     break;
 
                 case PROXY_WASM_HEADER_SCHEME:
-                    val = r->schema.data;
-                    val_len = r->schema.len;
+                    if (r->connection->ssl) {
```

Comment:
We can refactor the code and use getter instead?

Reply:
Do you mean to refactor the logic used above as well?

Reply:
do it later.

Reply:
will be done soon